sessions/2018/october/Session_15.md

```diff
@@ -0,0 +1,324 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 15 (16/10/2018)
+
+- Reviewing/fixing Spring API homework
+- API desing concepts
+- Dependency injection concepts
+
+### Notes:
+
+Roger was reviewing my homework solution and advicing me some modifications and improvements.
+
+#### Reviewing/fixing Spring API homework
+
+In certain part of the code I needed to save and incremental `id` `int` field into the object's field that was a `String`. A way to do it should be changing the type of the object's field, but a simpler way (hack) to work around it is doing `id` increment and mapping to `String` by concatenating with `""` at the moment to set into the object's field:
+
+```java
+book.setBookID(""+id++);
```

Comment:
Mini-interview question: What's the difference between `++id` and `id++`?

Reply:
- `++id` is the pre increment operator that increments the variable value just in the same step.
So:
```java
int number = 1;
System.out.println("++number = " + ++number);
```
Will give as result  `2`

- `id++` is the post increment operator, so this will increment variable just after the current step. Therefor:
```java
int number = 1;
System.out.println("number++ = " + number++);
```
Will give as result  `1`. But will be `2` in the next step, so for example, the next step being:
```java
System.out.println("number++ = " + (number + 10));
```
Will print out the number `12`